hbase-client/src/main/java/org/apache/hadoop/hbase/client/ReversedClientScanner.java

```diff
@@ -64,7 +64,9 @@ protected boolean setNewStartKey() {
 
   @Override
   protected ReversedScannerCallable createScannerCallable() {
+    // In case of meta, we want to support scan over a replica region.
+    int replicaId = (scan.getConsistency() == Consistency.TIMELINE) ? scan.getReplicaId() : 0;
```

Comment:
This is a bit strange, TIMELINE means we should send request to primary first and if it does not come back in time, we switch to secondary replicas. IIRC the design here is that, if targetReplicaId is -1, we follow the rule of STRONG or TIMELINE, if targetReplicaId is not -1, we will go to the specific replica directly?

Reply:
There is  no -1 involved. The flow is like that:
1. If it is Consistency.STRONG, no replica involved.
2. If it is Consistency.TIMELINE, and targetReplicaId is 0, it is the hedged read behavior (primary first, then go to replicas if no response within configured interval).
3. If it is  Consistency.TIMELINE, and targetReplicaId is a valid replica id, go to the target replica region within the specified replica id.

Reply:
The previous comment is incorrect, let me visit it again. Just found out the targetReplicaId is initialized to -1. Testing also fails, let me debug first.

Reply:
Just found out that what you said is correct.

Reply:
However, I introduced a bug here. For Consistency.TIMELINE, if scan is targeted to a specific replica region, replicaId is -1, which is incorrect. For this case, it also needs to pass in 0.

Reply:
For testing failures, it is my misunderstanding of the default targetReplicaId is 0 (actually it is -1). Introduce a utility method to differentiate the case for scan over a specific replica region in TIMELINE mode.